src/apps/loyalty/server/middlewares/user.ts

```diff
@@ -2,17 +2,20 @@ import { NextFunction, Request, Response } from "express"
 import { fetchCollectorProfile } from "../gravity"
 
 export default function UserMiddleware(req: Request, res: Response, next: NextFunction) {
-  if (req.user) {
-    fetchCollectorProfile(req.user.get("accessToken"))
-      .then(data => {
-        req.user.set("profile", data)
+  const { CURRENT_USER } = res.locals.sd
+  if (CURRENT_USER) {
+    fetchCollectorProfile(CURRENT_USER.accessToken)
+      .then(profile => {
+        CURRENT_USER.profile = profile
       })
       .then(next)
       .catch(err => {
-        req.user.set("profile", {
-          loyalty_applicant_at: "",
+        // TODO Do these really need to be empty strings?
```

Comment:
As the comment says, do we really need these to be empty strings? Instead of, say, `null`/`undefined`, or even leaving `profile` as `null`/`undefined`?